### PR TITLE
Query : GetBranches

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Adds support for GetBranches

## Changes
Adds support for GetBranches

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+import lombok.*;
+
+import java.util.List;
+
+@Data
+@ToString
+@RequiredArgsConstructor
+public class Branch {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:14]: Please doc this. 
E.g. "The branches above the last finalized block."

File: README.md [hunk 2]
```diff
@@ -373,6 +383,11 @@ val peers = client.getPeerList(true);
 SemVer version = client.getVersion();
 ```
 
+#### getBranches
+```java
+val branch = client.getBranches();
```
Comment (EmilLa1) [inline README.md:388]:
```suggestion
Branch branch = client.getBranches();
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    /**
+     * Hash of the block at the HEAD of this branch.
+     */
+    private Hash blockHash;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:19]: Please make the fields `final`.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.concordium.sdk.serializing.JsonMapper;
+import com.concordium.sdk.transactions.Hash;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+import lombok.*;
+
+import java.util.List;
+
+@Data
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:11]: I'm not sure why you also annotate with `@ToString` and `@RequiredArgsConstructor` as `@Data` already puts these on the type.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+     * Children of the {@link Branch#blockHash}.
+     */
+    private List<Branch> children;
+
+    /**
+     * Parses {@link concordium.ConcordiumP2PRpc.JsonResponse} to {@link Branch}
+     * @param branches {@link concordium.ConcordiumP2PRpc.JsonResponse}
+     * @return Parsed {@link Branch}
+     */
+    public static Branch fromJson(ConcordiumP2PRpc.JsonResponse branches) {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:31]: Please provide a unit test.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good, minor comments.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java (hunk 1 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:20]: done

File: README.md (hunk 2 above)
Comment (parv0888) [reply README.md]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java]: removed

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/branch/Branch.java:46]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 6]
```diff
@@ -2,6 +2,14 @@
 
 import com.concordium.sdk.exceptions.*;
 import com.concordium.sdk.responses.AccountIndex;
+import com.concordium.sdk.responses.branch.Branch;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:5]: The exceptions are already imported via the * above. Same counts for 
```
import com.concordium.sdk.responses.blocksatheight.BlocksAtHeight;
import com.concordium.sdk.responses.peerStats.PeerStatistics;
import com.concordium.sdk.responses.peerlist.Peer;
```

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good. 👍  Please remove obsolete imports and and rebase.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 6 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java]: done

**Review by EmilLa1 — APPROVED** [review]
Looks good

**Pull request merged**